### PR TITLE
[JUJU-4050] Cache watchable DB

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following uses the internal worker runner as a cache for accessing the underlying watchable DB. This removes an intermediate type as we can just expose the underlying watchable DB type.

When creating the Watchable DB we already have all the pieces in play, so we just need to align those pieces to the WatchableDB interface. Once they're aligned we can just return that.

This means that calling `WatchableDB` is really cheap as we're not calling `GetDB` unless we're using a new namespace.

We can do this because the underlying TxnRunner returned from GetDB actually wraps the db, so the db can be refreshed without refreshing the TxnRunner. If the TxnRunner fails, that will cause everything to come down anyway.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```sh
$ make juju jujud
$ juju bootstrap lxd test --build-agent
$ juju enable-ha
```

### CODE REVIEW

File: worker/changestream/worker_test.go [hunk 1]
```diff
@@ -107,10 +106,9 @@ func (s *workerSuite) TestEventSourceCalledTwice(c *gc.C) {
 
 	done := make(chan struct{})
 
-	s.dbGetter.EXPECT().GetDB("controller").Return(s.TxnRunner(), nil).Times(2)
-	s.eventMuxWorker.EXPECT().EventSource().Return(s.eventSource).Times(2)
-	s.eventMuxWorker.EXPECT().Kill().AnyTimes()
-	s.eventMuxWorker.EXPECT().Wait().DoAndReturn(func() error {
+	s.dbGetter.EXPECT().GetDB("controller").Return(s.TxnRunner(), nil).Times(1)
+	s.watchableDBWorker.EXPECT().Kill().AnyTimes()
```
Comment (SimonRichardson) [inline worker/changestream/worker_test.go:110]: Notice we only call the GetDB once now.

File: worker/changestream/manifold.go [hunk 2]
```diff
@@ -127,7 +127,7 @@ func changeStreamOutput(in worker.Worker, out interface{}) error {
 		var target changestream.WatchableDBGetter = w
 		*out = target
 	default:
-		return errors.Errorf("out should be a *changestream.ChangeStream; got %T", out)
+		return errors.Errorf("out should be a *changestream.WatchableDBGetter; got %T", out)
```
Comment (SimonRichardson) [inline worker/changestream/manifold.go:130]: driveby.

File: worker/changestream/worker.go [hunk 3]
```diff
@@ -131,13 +122,19 @@ func (w *changeStreamWorker) Report() map[string]any {
 
 // GetWatchableDB returns a new WatchableDB for the given namespace.
 func (w *changeStreamWorker) GetWatchableDB(namespace string) (changestream.WatchableDB, error) {
-	db, err := w.cfg.DBGetter.GetDB(namespace)
-	if err != nil {
-		return nil, errors.Trace(err)
+	// If the worker already exists, return the existing worker early.
+	if mux, err := w.runner.Worker(namespace, w.catacomb.Dying()); err == nil {
```
Comment (manadart) [inline worker/changestream/worker.go:126]: You specifically check for `ErrDead` in the db-accessor worker. Do you need to do that here?

I think both here and in that worker we should be proceeding on to `StartWorker` only if we got a `NotFound` error.

Comment (SimonRichardson) [reply worker/changestream/worker.go]: I've just normalized the two, so the behave in the same way.

Comment (SimonRichardson) [reply worker/changestream/worker.go]: We could make a generic function from this, but it seems a bit overkill imo.
